### PR TITLE
Added in support for allow-update for zones

### PR DESCRIPTION
**Pull request by fhaynes**
I needed support for allowing a zone to be updated dynamically, so I put this in. I made sure that the module will not overwrite a file while it is allowing updates from an IP, as that would overwrite the dynamic leases. Also preventing it from bumping the zone serial while it is accepting updates.

### CODE REVIEW

File: manifests/zone.pp [hunk 1]
```diff
@@ -13,6 +13,7 @@
   $zone_type = 'master',
   $allow_transfer = [],
   $allow_forwarder = [],
+  $allow_update = undef,
```
Comment (solarkennedy) [inline manifests/zone.pp]: Can you do a plain bool here?

File: manifests/zone.pp [hunk 2]
```diff
@@ -46,13 +47,24 @@
   } else {
     # Zone Database
 
+    # If it is a dynamic zone, we don't want to overwrite the existing zone file as it might
+    # destroy records that have been added
+    if $allow_update {
```
Comment (solarkennedy) [inline manifests/zone.pp]: Can you validate_bool() on this before you use it?

File: manifests/zone.pp (hunk 1 above)
Comment (solarkennedy) [inline manifests/zone.pp]: It looks like we haven't been adding docstrings to this type.

Can you start with
```
# == Parameters
#
# [*allow_update*]
# ...
#
define dns::zone...
```

Comment (fhaynes) [inline manifests/zone.pp]: If I do a bool there, I will need a second parameter to hold the list of IPs allowed to update. validate_bool() will fail if an array is passed to it, I think. If the second variable is acceptable, I'll push a commit with that change.

Comment (solarkennedy) [inline manifests/zone.pp]: Ah, gotcha. Eh this is fine.

File: manifests/zone.pp (hunk 2 above)
Comment (solarkennedy) [inline manifests/zone.pp]: Then what type are you expecting after the if block? Can you validate then?

Comment (solarkennedy) [inline manifests/zone.pp]: Array I suppose, right.

**Comment by solarkennedy** [comment]
I'm sorry this got lost so long ago. If this is still a feature you need, please rebase and ping me so I can merge it properly without getting lost to the sands of time.

**Pull request closed without merging**